sda/conode.go

```diff
@@ -101,3 +97,26 @@ func (c *Conode) Address() network.Address {
 func (c *Conode) GetService(name string) Service {
 	return c.serviceManager.Service(name)
 }
+
+// ProtocolRegisterName will sign up a new protocol to this Conode.
+// It returns the ID of the protocol.
+func (c *Conode) ProtocolRegisterName(name string, protocol NewProtocol) ProtocolID {
```

Comment:
The protocols can register themselves only in the `init()`-method, so we need a global registration (renaming it using `GlobalProtocolRegisterName`).